onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from onnxscript import ir
+from onnxscript.version_converter._adapter_lib import pick_adapter_set
+
+
+class _VersionConverter:
+    def __init__(self, model: ir.Model) -> None:
+        self._functions = model.functions
+        self.model_version = model.opset_imports.get("")
+        self.custom_adapters = {}
+
+    def _register_custom_adapters(self, custom_adapter_list) -> None:
+        self.custom_adapters = custom_adapter_list
+
+    def graph_version_convert(self, graph: ir.Graph, target_version: int) -> None:
+        if self.model_version == target_version:
+            # No conversion needed
+            return
+
+        # Iterate through all nodes in the graph
+        # Set node.version as the model_version
+        # TODO: Do this internally in IR?
+        for node in graph:
+            node.version = self.model_version
+
+        # Iterate from current model version -> target version
+        # Updating each node based on the correct adapter [ver->ver+1]
+        for opset_version in range(self.model_version, target_version):
```

Comment:
May be better to explicitly check for target_version being > or < current version. (It's ok to focus on up-conversion in first version, but may be better to have an error/warning message if we run into down-conversion when it is unimplemented.)

Reply:
Actually, this could be bundled into a check after `pick_adapter_set` to more generally handle the case when we don't have an adapter set (for either 23 to 24 or for 18 to 17).